lib/rdf/util/coercions.rb

```diff
@@ -0,0 +1,52 @@
+module RDF
+  module Util
+    module Coercions
+      # This is a provisional module intended to house input
+      # coercions. Currently the only coercion is a statement
+      # preprocessor that is used in e.g. {RDF::Writable#insert} and
+      # {RDF::Mutable#delete}.
+
+      ##
+      # Coerce a set of arguments into {RDF::Statement} objects and then
+      # operate over them with a block.
+      # 
+      # @example
+      #  process_statements(statements) { |value| do_something(value) }
+      #
+      # @param statements [#map] The arbitrary-ish input to be manipulated
+      # @param query [false, true] Whether to call +query+ before the block
+      # @param constant [false, true] Whether to test if the statements
+      #  are constant
+      # @yield [RDF::Statement, RDF::Enumerable] 
+      # @return statements
+      def process_statements(statements, query: false, constant: false, &block)
+        raise ArgumentError, 'expecting a block' unless block_given?
```

Comment:
Given the usage, perhaps `#coerce_statements` would be more appropriate

Reply:
again not married to any particular terminology